### ISSUE TITLE
player.allSpotted is returning dead players

### ISSUE THREAD

**Issue by billfreeman44** [saul/demofile#69]
As title says, allSpotted is returning dead people.  You can check this using this MWE

```js
demoFile.gameEvents.on('weapon_fire', e => {
	let spottedList=player.allSpotted;
	for (i=0; i<spottedList.length; i++) {
		if (spottedList[i].isAlive == true){
			console.log('ALIVE '+spottedList[i].name)
		} else {
			console.log('DEAD '+spottedList[i].name)
		}
	}
});
```

I think a possible fix might be to add an isAlive check to the allSpotted function like this: 

```js
  get allSpotted() {
    return this._demo.players
      .filter(p => p.clientSlot !== this.clientSlot && this.hasSpotted(p) && p.isAlive == true);
  }
```

**Comment by necuk** [comment]
@billfreeman44 here is how I deal with that:

function isIngame(player) {
   return (player.teamNumber == 2 || player.teamNumber == 3)
}

if ((!spotted.isFakePlayer) && (spotted.isAlive) && (isIngame(spotted)))
    do smth

**Comment by saul** [comment]
That's definitely strange that dead players can be 'spotted'. I wonder if that's just a strange bug in the game - but yes I agree that they should be filtered out.

@necuk's solution would work. But it's worth pointing out that the `isIngame` check is unnecessary - a player can't be alive but spectating (team #1) or unassigned (team #0)

P.S. I hope you both don't mind that I edited your comments to add syntax highlighting.

**Comment by saul** [comment]
Fixed in 0.4.17.
